dart/button_with_closure.dart

```diff
@@ -0,0 +1,9 @@
+import 'dart:html';
+void main() {
+  for (var i = 0; i < 10; i++) {
+    var button = new ButtonElement();
+    button.text = i;
+    button.onClick.listen(((i) => (e) => window.alert(i))(i));
```

Comment:
Kind of useless because dart already evaluates variables when assigned rather than when referenced...